test/test_fit_transform.py

```diff
@@ -96,9 +96,7 @@ def test_lfda(self):
     res_2 = lfda.fit_transform(self.X, self.y)
 
     # signs may be flipped, that's okay
-    if np.sign(res_1[0,0]) != np.sign(res_2[0,0]):
-        res_2 *= -1
-    assert_array_almost_equal(res_1, res_2)
+    assert_array_almost_equal(abs(res_1), abs(res_2))
```

Comment:
The tests were failing because of a sign error, I think it's because of the eigenvalue decomposition in LFDA, that can lead to different signs in the output. I did this modif because I thought that any sign switching of any column still keeps the result valid am I right (any symmetry wrt some of the axis are still valid right?) (Not just a global sign switching as before)? But maybe I should rather try to change the code to make LFDA deterministic ? What do you think @perimosocordiae @bellet ?

Reply:
I just raised an issue for that (#211),  I guess maybe let's do like that for now and later fix this non deterministic problem ?

Reply:
The sign switch is expected, as each eigenvector may be negated independently of the others. The result is valid in the sense that the underlying optimization problem is still solved, and distances in the transformed space are equivalent. Individual points in the transformed space won't always be the same, though.

I don't think it's worth trying to enforce a deterministic result, but I could probably be persuaded otherwise.

Reply:
Let's discuss it in the issue you opened